core/src/main/java/com/yahoo/ycsb/generator/IncrementingPrintableStringGenerator.java

```diff
@@ -0,0 +1,388 @@
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.LETTER_NUMBER);
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.OTHER_NUMBER);
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.UPPERCASE_LETTER);
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.LOWERCASE_LETTER);
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.TITLECASE_LETTER);
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.MODIFIER_LETTER);
+    CHAR_TYPE_EXTENDED_ALPHANUMERICS.add((int)Character.OTHER_LETTER);
+  }
+  
+  /** The character set to iterate over. */
+  private final int[] characterSet;
+  
+  /** An array indices matching a position in the output string. */
+  private int[] indices; 
+  
+  /** The length of the output string in characters. */
+  private final int length;
+  
+  /** The last value returned by the generator. Should be null if {@link #nextValue()} 
+   * has not been called.*/
+  private String lastValue;
+  
+  /** Whether or not to throw an exception when the string rolls over. */
+  private boolean throwExceptionOnRollover;
+  
+  /** Whether or not the generator has rolled over. */
+  private boolean hasRolledOver;
+  
+  /**
+   * Generates strings of 8 characters using only the upper and lower case alphabetical
+   * characters from the ASCII set. 
+   */
+  public IncrementingPrintableStringGenerator() {
+    this(DEFAULTSTRINGLENGTH, printableBasicAlphaASCIISet());
+  }
+  
+  /**
+   * Generates strings of {@link #length} characters using only the upper and lower 
+   * case alphabetical characters from the ASCII set. 
+   * @param length The length of string to return from the generator.
+   * @throws IllegalArgumentException if the length is less than one.
+   */
+  public IncrementingPrintableStringGenerator(final int length) {
+    this(length, printableBasicAlphaASCIISet());
+  }
+  
+  /**
+   * Generates strings of {@link #length} characters using the code points in
+   * {@link #characterSet}.
+   * @param length The length of string to return from the generator.
+   * @param characterSet A set of code points to choose from. Code points in the 
+   * set can be in any order, not necessarily lexical.
+   * @throws IllegalArgumentException if the length is less than one or the character
+   * set has fewer than one code points.
+   */
+  public IncrementingPrintableStringGenerator(final int length, final int[] characterSet) {
+    if (length < 1) {
+      throw new IllegalArgumentException("Length must be greater than or equal to 1");
+    }
+    if (characterSet == null || characterSet.length < 1) {
+      throw new IllegalArgumentException("Character set must have at least one character");
+    }
+    this.length = length;
+    this.characterSet = characterSet;
+    indices = new int[length];
+  }
+  
+  @Override
+  public String nextValue() {
+    if (hasRolledOver && throwExceptionOnRollover) {
+      throw new RuntimeException("The generator has rolled over to the beginning");
```

Comment:
could we make this a subclass of RuntimeException so that if folks want to catch it for whatever reason they needn't parse the message.

alternatively, maybe we should add an accessor for hasRolledOver in case a user need to know if things have rolled over, but doesn't want to consider it an error?